src/orion/benchmark/benchmark_client.py

```diff
@@ -11,13 +11,19 @@
 from orion.benchmark.task.base import bench_task_factory
 from orion.core.io.database import DuplicateKeyError
 from orion.core.utils.exceptions import NoConfigurationError
-from orion.storage.base import get_storage, setup_storage
+from orion.storage.base import setup_storage
 
 logger = logging.getLogger(__name__)
 
 
 def get_or_create_benchmark(
-    name, algorithms=None, targets=None, storage=None, executor=None, debug=False
+    name,
+    algorithms=None,
+    targets=None,
+    storage=None,
+    executor=None,
+    debug=False,
+    storage_instance=None,
```

Comment:
Documentation missing for storage_instance. Also the user must now pass at least one of them, or is setup_storage using global config otherwise?

Reply:
I don't like it much because it makes it hard to track the number of storage instantiated.
I think the storage should be the core of the Orion client instead of the experiment

Reply:
You mean that it is hard to figure out how many storage objects have been created through `create_experiment/benchmark` calls with storage options? 

Yeah, it's not much more cumbersome to just wrap the config in another function to create the storage. 
```
benchmark = create_benchmark(storage=setup_storage(storage_config))
```
I would agree to keep storage only. We'd need to support storage as a configuration too during a deprecation phase however.

Reply:
Done, thanks!